posthog/migrations/0126_stricter_team.py

```diff
@@ -0,0 +1,57 @@
+# Generated by Django 3.0.6 on 2021-02-09 09:11
+
+import django.core.validators
+import django.db.models.deletion
+from django.db import migrations, models
+
+import posthog.models.utils
+
+
+def prepare_teams(apps, schema_editor):
+    Team = apps.get_model("posthog", "Team")
+    Team.objects.filter(organization_id__isnull=True).delete()
+    Team.objects.filter(opt_out_capture=True).update(anonymize_ips=True)
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("posthog", "0125_longer_webhook_url"),
+    ]
+
+    operations = [
+        migrations.RunPython(prepare_teams, migrations.RunPython.noop),
+        migrations.AlterField(
+            model_name="team",
+            name="api_token",
+            field=models.CharField(
+                default=posthog.models.utils.generate_random_token,
+                max_length=200,
+                unique=True,
+                validators=[
+                    django.core.validators.MinLengthValidator(
+                        10, "Project's API token must be at least 10 characters long!"
+                    )
+                ],
+            ),
+        ),
+        migrations.AlterField(
+            model_name="team",
+            name="name",
+            field=models.CharField(
+                default="Default Project",
+                max_length=200,
+                validators=[django.core.validators.MinLengthValidator(1, "Project must have a name!")],
+            ),
+        ),
```

Comment:
Should ensuring this validates also go into the `prepare_teams` block?

Reply:
As far as I understand, this won't affect existing column values, and we regenerate the value with a proper function, so this is just more like a guard

Reply:
If so, wouldn't that then lead to random errors whenever `team.save()` is called anywhere? I'm not sure if this would be triggered or not.

Reply:
Added generation to `prepare_team` just in case